### ISSUE TITLE
Question regarding `pred_logits`

### ISSUE THREAD

**Issue by anwai98** [microsoft/BiomedParse#46]
Hi team,

First of all, thanks for the cool work!

I have a detailed question out of curiousity: what are the `pred_logits` in results and what's the variable assigned as `pred_class` used for? (https://github.com/microsoft/BiomedParse/blob/main/inference_utils/inference.py#L71)

Thanks in advance!

**Comment by theodore-zhao** [comment]
The model has a pre-defined list of meta object types. When performing text prompt segmentation, the model outputs one of the classes in meta object types based on semantic similarity with the prompt. The predicted class is an estimate of the meta object type for the prompt in parallel to the predicted segmentation mask.

**Comment by anwai98** [comment]
Hi @theodore-zhao,

Best wishes for the New Year!

> The predicted class is an estimate of the meta object type for the prompt in parallel to the predicted segmentation mask.

I see. Makes sense. I saw the meta class outputs mapped to an unused variable which confused me. Thanks for the clarification.

Another question: can you tell me if it is possible to perform semantic segmentation without prompts? (say, I put an image in the model, and it detects and outputs all available semantic classes). From what I understand from the paper (correct me if I am wrong), it is possible to do so. Could you hint me to the scripts for this?

Thanks in advance!

**Comment by theodore-zhao** [comment]
Yes. Basically what you do is to iterate through the targets for the specific modality (e.g. CT-Abdomen) and prompt the model for a set of segmentation masks (you can pre-filter by p-values). In order to resolve potential overlapping, you can use https://github.com/microsoft/BiomedParse/blob/main/inference_utils/output_processing.py#L45 to combine the masks into disjoint ones. 

Let me know if this works for you!

**Comment by anwai98** [comment]
Thanks for your response.
I think I vaguely understand the workflow. Could you help me with a sample script similar to the one for prompt-based inference [here](https://github.com/microsoft/BiomedParse/blob/main/inference_utils/inference.py#L38) to reproduce the pointers you mentioned?

Would be nice to get a script such as:
```python
input_array = ...  # this would be an input (2d) array
segmentation = automatic_segmentation(image=input_array)  # where, segmentation is the "automatic" / prompt-free outputs from the model
```

Thanks in advance!

**Comment by theodore-zhao** [comment]
I just added an example notebook for your need at https://github.com/microsoft/BiomedParse/blob/main/inference_examples_recognition.ipynb

Let me know if that works for you.

**Comment by anwai98** [comment]
Hi @theodore-zhao,

I tried it out on a few images. It does look as expected to me. Thank you so much for releasing an overview notebook and the functionality for it!

And thank you so much for all your help!

PS. I might come back with some other questions!
